app/code/community/Nexcessnet/Turpentine/misc/version-4.vcl

```diff
@@ -310,6 +310,16 @@ sub vcl_hash {
             req.http.Cookie ~ "customer_group=") {
         hash_data(regsub(req.http.Cookie, "^.*?customer_group=([^;]*);*.*$", "\1"));
     }
+
+    # Some extension controllers serve different results for a single url if the
+    # request is xhr. both prototype and jquery set X-Requested-With headers
+    # on ajax requests so we can add that to the hash so that both variations
+    # are served seperately
+    if (req.http.X-Requested-With) {
+      std.log("hash_data xhr request - X-Requested-With: " + req.http.X-Requested-With);
```

Comment:
Do you think we should leave this here? Maybe wrap it with a debug condition might be better,